theano/sandbox/cuda/dnn.py

```diff
@@ -2321,10 +2321,10 @@ def local_avgpool_dnn_grad_stride(node):
                 (inp_grad.owner and isinstance(inp_grad.owner.op,
                                                HostFromGpu))):
                 desc = GpuDnnPoolDesc(ws=ds, stride=st, mode=mode, pad=pad)()
+                contiguous_inp_grad = gpu_contiguous(inp_grad)
                 ret = GpuDnnPoolGrad()(gpu_contiguous(inp),
-                                       gpu_contiguous(numpy.empty((1,1,1,1),
-                                           dtype=numpy.float32)),
-                                       gpu_contiguous(inp_grad),
+                                       contiguous_inp_grad,
```

Comment:
While this might respect the expected shape, why would you compute the right values?

This is supposed to be the output of the GpuDnnPool operation and we have no indication that the value is not used by the operation.  So why are we trying to do some fragile replacement like this?

Note that I didn't review the original code or I would have had the same comment.